.github/workflows/ci_workflows.yml

```diff
@@ -18,56 +22,35 @@ jobs:
             python: 3.x
             toxenv: codestyle
 
-          - name: Python 3.8 with minimal dependencies
+          - name: Python 3.10 with astropy data and coverage
             os: ubuntu-latest
-            python: 3.8
-            toxenv: py38-test
+            python: '3.10'
+            toxenv: py310-test-cov
+            toxargs: -v
+            toxposargs: --open-files --remote-data=astropy
 
-          - name: Python 3.9 with all optional dependencies
-            os: ubuntu-latest
-            python: 3.9
-            toxenv: py39-test-alldeps
-            toxargs: -v --develop
-            toxposargs: --open-files
-
-          # NOTE: In the build below we also check that tests do not open and
-          # leave open any files. This has a performance impact on running the
-          # tests, hence why it is not enabled by default.
-          - name: Python 3.8 with oldest supported version of all dependencies
+          - name: Python 3.8 with oldest supported version of key dependencies
             os: ubuntu-18.04
             python: 3.8
             toxenv: py38-test-oldestdeps
 
-          - name: Python 3.8 with numpy 1.17 and full coverage
```

Comment:
Numpy 1.17 pin here is also a lie. You require astropy>=5.1 which means you also require numpy>=1.18